src/dmd/expressionsem.d

```diff
@@ -8472,6 +8472,19 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                     // OR it in, because it might already be set for C array indexing
                     exp.indexIsInBounds |= bounds.contains(getIntRange(exp.e2));
                 }
+                else if (auto ve = exp.e1.isVarExp())
+                {
+                    if (sc.flags & SCOPE.Cfile && t1b.ty == Tsarray)
```

Comment:
Micro-optimization: swap this condition with the one above. Compiling C files occurs less frequent than compiling D files.

Also, I am a bit worried for the effect that all of these C file conditions have on the compilation of normal D code.

Reply:
ok.

I work to minimize these changes.